runtime/gc_glue_java/ArrayletObjectModel.hpp

```diff
@@ -1039,7 +1031,11 @@ class GC_ArrayletObjectModel : public GC_ArrayletObjectModelBase
 		} else if (dataSizeInBytes < _omrVM->_arrayletLeafSize) {
 			isValidDataAddress = (dataAddr == (void *)((uintptr_t)arrayPtr + contiguousIndexableHeaderSize()));
 		} else {
-			isValidDataAddress = (dataAddr == NULL);
+			if (isVirtualLargeObjectHeapEnabled()) {
```

Comment:
this part seems unnecessary is we really use this method in offheap enabled case